packages/next-app/src/components/Link.tsx

```diff
@@ -0,0 +1,30 @@
+"use client";
+
+import { Url } from "next/dist/shared/lib/router/router";
+import Link from "next/link";
+import React from "react";
+
+import { Spinner, SpinnerColor } from "./Spinner";
+
+export function LinkComponent({
+  href,
+  content,
+  loaderColor = "blue",
+}: {
+  href: Url;
+  content: React.ReactElement;
+  loaderColor?: keyof typeof SpinnerColor;
+}) {
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  const ClonedElement = React.cloneElement(content, {
+    onClick: () => setIsLoading(true),
+    children: isLoading ? (
+      <Spinner color={loaderColor} size={"sm"} />
+    ) : (
+      content.props.children
+    ),
+  });
```

Comment:
Saw what you wanted to do here. You could also have [enforced a single child](https://react.dev/reference/react/Children#children-only) and skipped the added complexity here, wdyt? Am I missing something here?

Reply:
not quite if I understood 😅

let's take this example: 

```js

    <LinkComponent
      href={`/internalmanager/${network}/withdraw/all`}
      content={
        <Button
          className="flex items-center gap-1"
          shade="light"
          variant="outline"
          title="Withdraw all"
          disabled={!hasTokenWithBalance}
        >
          <MinusIcon />
          Batch Withdraw
        </Button>
      }
    />
``` 

here I'm receiving `<Button>` and I want to keep everything the same but the inner text inside this button, which would be its child: `<MinusIcon /> Batch Withdraw`. So I'm using `cloneElement` to handle the `isLoading` state and change the inner Button text when true. From what I read in the documentation I cannot achieve the same with enforcing a single child. If it's possible can you give more details on that?

Reply:
That makes sense! I had in mind something that I tried implementing and landed in a very similar to what you had here 😝 your implementation is more direct though.